dependent-service-downtime-and-ps-adaptor-behavior.md

```diff
@@ -0,0 +1,64 @@
+## Service Failures and Expected Behavior
+
+This guide outlines the expected behavior of the PS Adaptor when various dependent services are down. 
+The PS Adaptor works in conjunction with several services including the GP2GP Translator, MHS Adaptor, Facade, and others. 
+The guide explains the effects of outages and how the system behaves in each case.
+
+### System Overview
+The PS Adaptor initiates a conversation for transferring patient records, relying on several interconnected services:
+- GP2GP Translator: Sends patient records to the PS Adaptor via a queue.
+- Facade Application: Works with the PS Adaptor to manage transfer requests.
+- MHS Adaptor: Manages inbound and outbound queues, which are responsible for transferring patient records to the incumbent system.
+- Other Dependencies: Include PS Adaptor database, MHS inbound and outbound services, file storage, Redis cache, and MHS outbound SDS.
+
+### Service Outage Scenarios and Expected Behavior
+1. GP2GP Translator is Down
+   - Scenario: The GP2GP Translator is not operational and cannot send patient records.
+   - Expected Behavior: No messages are sent to the incumbent system while the translator is down. Once the GP2GP Translator recovers, 
+     the transfer process resumes, and messages are processed as normal.
+   - Recovery: Automatic upon Translator recovery; transfer resumes without manual intervention.
+2. Queue is Down
```

Comment:
😕 Not sure if a "queue" can go down. Is this more the scenario that the [broker](https://en.wikipedia.org/wiki/Message_broker) goes down?

Reply:
Yes, agreed as it is more likely scenario and more often it is a message broker itself that is down rather than a queue, however there are some scenarios when the queue itself can go down while the message broker could be still running and operational. For example, if the queue reaches its maximum capacity based on some config it becomes unreachable. There are also some other failures that can happen to the queue.

Reply:
Hmm, I'd recommend being specific about the particular scenario actually tested.

So if we've tested the scenario of "Broker dies", then let's document the behaviour then. If we've tested the scenario of "Broker alive, but not able to save the message", then let's document that behaviour.

For any other scenarios which aren't tested, let's not have them listed here.

Reply:
in NIAD-2369 in the file attached for tests it says "Queue" but I guess it meant to be more like the broker is down.